server/monitor-types/real-browser-monitor-type.js

```diff
@@ -253,6 +253,40 @@ class RealBrowserMonitorType extends MonitorType {
             timeout: monitor.interval * 1000 * 0.8,
         });
 
+        // Check for keyword if configured
+        if (monitor.keyword && monitor.keyword.trim()) {
+            try {
+                // Extract all visible text content from the page
+                let textContent = await page.textContent("body");
+
+                if (textContent) {
+                    textContent = textContent.replace(/\s+/g, " ").trim();
```

Comment:
Why this?
Could you add a comment?

Reply:
html collapes duplicate spaces <div>Hello   my  <p>     world</p></div> would just read "hello my world" when rendered in html. maybe textContent would already handle that, but making sure its "hello my world" and not "Hello   my       world" makes it more predictable if the user is copying some phrase to match out of the browser instead of using the inspector to get the actual raw spacing.

maybe comment should be "remove duplicate whitespace as html also collapses them". Would that make more sense?

Reply:
I have added more documentation and also made the tests cover this case.

I also made sure that the keywords is processed in the same way, so if a user enters the keyword with duplicate spaces, it will get normalized too and will match the normalized text content. a spaces-only keyword will be handled as no keyword (disabling the feature).

The handling of special characters is depending on the textContent function. I tested it with `&nbsp;` and it is threadened as whitespace while `&#9;` is not.

The "normalize whitespace" feature is about making it easier to match a multi word keywords without needing to worry about random spaces in the html. The user can enter the keyword with or without extra spaces and it will match anyways.